decidim-admin/app/permissions/decidim/admin/permissions.rb

```diff
@@ -64,6 +64,31 @@ def read_admin_dashboard_action?
         toggle_allow(user.admin? || space_allows_admin_access_to_current_action?)
       end
 
+      def admin_newsletter_action?
+        return unless permission_action.subject == :newsletter
+        return allow! if user.admin?
+        return unless space_allows_admin_access?
+        newsletter = context.fetch(:newsletter, nil)
+
+        case permission_action.action
+        when :index, :create
+          return allow!
+        when :read, :update, :destroy
+          return toggle_allow(user == newsletter.author)
+        end
+      end
+
+      def space_allows_admin_access?
+        user_has_admin_access = false
+        Decidim.participatory_space_manifests.each do |manifest|
+          Decidim.find_participatory_space_manifest(manifest.name)
+                 .participatory_spaces.call(organization)&.map do |space|
+            user_has_admin_access = true if space.admins.exists?(id: user.id)
```

Comment:
Improve readability? `user_has_admin_access = space.admins.exists?(id: user.id)`

Reply:
In this case, only when is `true` has to be assigned to the var, if `!true` must remain the previous value,
if not only the last iteration will have precedence, and set the value.

Reply:
if the comment below does what is inteded and improves performance, I'll go for it. :+1: